src/Umbraco.AuthorizedServices/Controllers/AuthorizedServiceController.cs

```diff
@@ -138,10 +142,32 @@ public IActionResult SaveToken(AddToken model)
         return Ok();
     }
 
+    /// <summary>
+    /// Generates access token for an authorized service.
+    /// </summary>
+    /// <param name="model">Request model identifying the service.</param>
+    /// <returns></returns>
+    [HttpPost]
+    public async Task<IActionResult> GenerateToken(GenerateToken model)
+    {
+        ServiceDetail serviceDetail = _serviceDetailOptions.Get(model.Alias);
+
+        Models.AuthorizationResult result = await _serviceAuthorizer
+            .AuthorizeServiceAsync(serviceDetail.Alias, string.Empty, string.Empty, string.Empty);
```

Comment:
It might be cleaner to add some method overloads here, rather than having to pass these empty strings.

So we extend `IAuthorizedServiceAuthorizer` to be:

- `AuthorizeOAuth2AuthorizationCodeServiceAsync(string serviceAlias, string authorizationCode, string redirectUri, string codeVerifier)`
- `AuthorizeOAuth2ClientCredentialsServiceAsync(string serviceAlias)`

And then similar with `IAuthorizationParametersBuilder`:

- `BuildParametersForOAuth2AuthorizationCode(ServiceDetail serviceDetail, string authorizationCode, string redirectUri, string codeVerifier)`
- `BuildParametersForOAuth2ClientCredentials(ServiceDetail serviceDetail)`

Reply:
Updated